analytics_data_api/middleware.py

```diff
@@ -14,6 +16,23 @@
     ReportFileNotFoundError,
 )
 
+thread_data = local()
+
+
+class RequestVersionMiddleware:
+    """
+    Add a database hint, analyticsapi_database, in the form of an attribute in thread-local storage.
+    This is used by the AnalyticsApiRouter to switch databases between the v0 and v1 views.
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if 'api/v1' in request.path:
+            thread_data.analyticsapi_database = getattr(settings, 'ANALYTICS_DATABASE_V1', None)
```

Comment:
it'd be nice to have an else clause to always set analyticsapi_database instead of having it be a maybe

Reply:
I agree with @ashultz0. We should handle the case the settings value is returning None.

Reply:
What should the `else` value be, in your opinion? `default`? If a Django router returns a `None` value from any of its four methods (e.g. `db_for_read`) then it automatically delegates to the `default` database. I don't want to make a decision in this middleware of what database to use in the else case. I think that the else case should be up to the router, not this middleware. I think this middleware's responsibility is to determine whether a particular database is required by the version of the API that is being used, but I don't think it should know what to select if the API is not being used. I would rather it fall back to the default behavior of the database router. What do you think?

Reply:
I see what you're saying but to me the mental model is the thread_data as set in the middleware is only the analyticsapi_database, it's not actually making a statement about how that is used. It's just saying this DB is the analytics DB for this URL. Analytics DB is V1 if it's in the URL, otherwise it's regular analytics DB.

later your router says hey I need analytics DB and the thread_data may have info for me

I don't feel very strongly about this, I just like values that are set. So up to you.

Reply:
No strong opinion here with me either. With your explanation, this looks good for me.

Reply:
I took in your feedback.